src/types/plugin-inputs.ts

```diff
@@ -11,14 +11,28 @@ export interface PluginInputs<T extends SupportedEventsU = SupportedEventsU, TU
   ref: string;
 }
 
-/**
- * This should contain the properties of the bot config
- * that are required for the plugin to function.
- *
- * The kernel will extract those and pass them to the plugin,
- * which are built into the context object from setup().
- */
-export const pluginSettingsSchema = T.Object({});
+export const pluginSettingsSchema = T.Object({
+  /**
+   * The amount of validations needed to consider a pull-request by a collaborator to be deemed eligible for merge
+   */
+  collaboratorMinimumApprovalsRequired: T.Number({ default: 1, minimum: 1 }),
+  /**
+   * The amount of validations needed to consider a pull-request by a contributor to be deemed eligible for merge
+   */
+  contributorMinimumApprovalsRequired: T.Number({ default: 1, minimum: 1 }),
+  /**
+   * The timespan to wait before merging a collaborator's pull-request
+   */
+  collaboratorMergeTimeout: T.String({ default: "3.5 days" }),
+  /**
+   * The timespan to wait before merging a contributor's pull-request
+   */
+  contributorMergeTimeout: T.String({ default: "7 days" }),
+  /**
+   * The location of the database
+   */
+  databaseUrl: T.String({ default: "database/sql.db" }),
```

Comment:
shouldn't this be in the env?

Reply:
I thought it would be better to have it here so it can be hosted anywhere (you can put an url for example)

Reply:
actually I'm worried about concurrency, like when two events trigger the plugin, both will take current DB file and add/remove PR and then make a commit but only one of them will successfully do it. 
how come you decided to use a local sqlite DB?